allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -23,7 +34,59 @@ def get_output_dim(self):
         return self.output_dim
 
     def forward(
-        self, token_ids: torch.LongTensor, mask: torch.LongTensor
+        self,
+        token_ids: torch.LongTensor,
+        mask: torch.LongTensor,
+        wordpiece_mask: torch.LongTensor = None,
+        offsets: torch.LongTensor = None,
     ) -> torch.Tensor:  # type: ignore
+        """
+        # Parameters
+
+        token_ids: torch.LongTensor
+            Shape: [batch_size, num_wordpieces].
+        mask: torch.LongTensor
+            Shape: [batch_size, num_orig_tokens if self._intra_word_tokenized else num_wordpieces].
```

Comment:
This is one reason why I might have split this into two classes instead of having it be a flag on one class.  I can see arguments both ways, but I'm somewhat partial to having separate collections of (Tokenizer, Indexer, Embedder) for each of the three cases we discussed, instead of different flags.

(Sorry, didn't make it to reviewing this PR today, just saw this discussion.  I'll try to get to it tomorrow, but no promises; Fridays are really busy for me.)

Reply:
In the beginning I didn't think it warrants a standalone class, not foreseeing a lot of complications. But it looks like that's necessary for this to be clean, so I will try to make that change now. I will take you both off the reviewer list for now and re-request review when I'm done.

Reply:
(If I can figure out how to move people off the reviewer list)

Reply:
Ah, the issue with `get_text_field_mask` makes sense. Thanks for explaining. I might wait for Matt to give his first pass before attempting to break apart the code. But if you can avoid duplicating the other logic, I certainly don't object.

And you can just ping us explicitly. No need to modify the reviewer list.